### DISCUSSION TITLE
JDL entity schema and field types explained

### DISCUSSION THREAD

**Discussion by simpletasks** [jhipster/generator-jhipster#26625] [Q&A]
Hi.
I want to understand filed types defined in JDL entities. Also I want to understand JDL schema specifically related to entities.
Where I can find JDL schema?
where I can find description of defined field types?
For particular database I can find great description of data types in online sources. I would  like to read something similar about JDL field types.
I try to map between JDL field types and database data types. Staring with JDL when generating JHipster version of  app JDL is mapped to Object types and Relational part is created in a form of Liquibase change sets.

**Comment by TatyOko28** [comment]
Hi.

If you want to understand **field types in JDL entities**, how JDL entities are structured, and how JDL types map to Java/database types, the best way is to look at both the documentation and how JHipster generates code internally.

## What is JDL?

JDL (**JHipster Domain Language**) is a DSL used to describe:

```text
- entities
- fields
- validations
- relationships
- enums
- pagination
- DTOs
- services
- filtering
- deployment options
